helper/tests/helper-tests-allavro/src/test/java/com/linkedin/avroutil1/compatibility/FieldBuilderTest.java

```diff
@@ -67,6 +67,39 @@ public void testDefaultValuesInterop() throws Exception {
         setDefaultValues(schema.getField("unionFieldWithDefaultRecord"));
     }
 
+    @Test
+    public void testDefaultSortOrderIsAscending() throws Exception {
+        // This record has no order specified for any of its fields. So, they should all be ASCENDING.
+        String avsc = TestUtil.load("RecordWithDefaults.avsc");
+        Schema schema = Schema.parse(avsc);
+        for (Schema.Field field : schema.getFields()) {
+            Assert.assertEquals(field.order(), Schema.Field.Order.ASCENDING);
+        }
+    }
+
+    @Test
+    public void testNewFieldSortOrder() throws Exception {
+        // default (no order specified).
+        Schema.Field field = AvroCompatibilityHelper.newField(null).setName("default").build();
+        Assert.assertEquals(field.order(), Schema.Field.Order.ASCENDING);
+
+        // explicit set the order to null (in our API); should get transformed to ASCENDING.
```

Comment:
should result in IllegalArgumentException out of build(). likely users who set to null are doing something wrong?

Reply:
Done, but doing it in setOrder(), as discussed offline.